service-loadbalancer/gce/controller.go

```diff
@@ -0,0 +1,295 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/experimental"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/client/record"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/golang/glog"
+)
+
+var (
+	keyFunc          = framework.DeletionHandlingMetaNamespaceKeyFunc
+	lbControllerName = "lbcontroller"
+)
+
+// loadBalancerController watches the kubernetes api and adds/removes services
+// from the loadbalancer, via loadBalancerConfig.
+type loadBalancerController struct {
+	client         *client.Client
+	ingController  *framework.Controller
+	nodeController *framework.Controller
+	svcController  *framework.Controller
+	ingLister      StoreToIngressLister
+	nodeLister     cache.StoreToNodeLister
+	svcLister      cache.StoreToServiceLister
+	clusterManager *ClusterManager
+	recorder       record.EventRecorder
+	nodeQueue      *taskQueue
+	ingQueue       *taskQueue
+	tr             *gceTranslator
+	stopCh         chan struct{}
+	// stopLock is used to enforce only a single call to Stop is active.
+	// Needed because we allow stopping through an http endpoint and
+	// allowing concurrent stoppers leads to stack traces.
+	stopLock sync.Mutex
+	shutdown bool
+}
+
+// NewLoadBalancerController creates a controller for gce loadbalancers.
+func NewLoadBalancerController(kubeClient *client.Client, clusterManager *ClusterManager, resyncPeriod time.Duration) (*loadBalancerController, error) {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(kubeClient.Events(""))
+
+	lbc := loadBalancerController{
+		client:         kubeClient,
+		clusterManager: clusterManager,
+		stopCh:         make(chan struct{}),
+		recorder: eventBroadcaster.NewRecorder(
+			api.EventSource{Component: "loadbalancer-controller"}),
+	}
+	lbc.nodeQueue = NewTaskQueue(lbc.syncNodes)
+	lbc.ingQueue = NewTaskQueue(lbc.sync)
+
+	// Ingress watch handlers
+	pathHandlers := framework.ResourceEventHandlerFuncs{
+		AddFunc:    lbc.ingQueue.enqueue,
+		DeleteFunc: lbc.ingQueue.enqueue,
+		UpdateFunc: func(old, cur interface{}) {
+			if !reflect.DeepEqual(old, cur) {
+				glog.Infof("Ingress %v changed, syncing",
+					cur.(*experimental.Ingress).Name)
+			}
+			lbc.ingQueue.enqueue(cur)
+		},
+	}
+	lbc.ingLister.Store, lbc.ingController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func() (runtime.Object, error) {
```

Comment:
ugh, this line breaking is ugly.  Can you extract this out as a separate function, instead of using an anonymous func